### PR TITLE
Re-fix Ensmallen History Mod Checker

### PR DESCRIPTION
**Pull request by birm**
I'm honestly not sure exactly why the @coatless method didn't work (#171 #172 ), but this is my attempt to fix it.

### CODE REVIEW

**Comment by birm** [comment]
Hopefully the third time is the charm!

**Comment by birm** [comment]
@coatless I can't seem to request you as a reviewer, but please take a look.

File: HISTORY.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
   * Require new HISTORY.md entry for each PR.
     ([#171](https://github.com/mlpack/ensmallen/pull/171),
-     [#172](https://github.com/mlpack/ensmallen/pull/172)).
+     [#172](https://github.com/mlpack/ensmallen/pull/172),
+     [#175](https://github.com/mlpack/ensmallen/pull/175)).
```
Comment (zoq) [inline HISTORY.md]: Pretty sure, this is the last time we have to add another PR number to the list.

Comment (coatless) [reply HISTORY.md]: Famous last words :)

File: scripts/history-update-check.sh [hunk 2]
```diff
@@ -6,11 +6,10 @@
 #   $ history-update-check.sh
 #
 # This should be run from the root of the repository.
-set -e
-
-! git diff --exit-code master -- HISTORY.md > /dev/null
-
-if [ $? ]; then
+res=$(git diff origin/master --name-only | grep ^HISTORY.md | wc -l)
```
Comment (coatless) [inline scripts/history-update-check.sh]: Nice workaround.

File: scripts/history-update-check.sh [hunk 3]
```diff
@@ -6,11 +6,10 @@
...
+echo "Files Changed:"
+git diff --name-only
```
Comment (coatless) [inline scripts/history-update-check.sh]: Is this to just show what changed in files with the CI?

Comment (birm) [reply scripts/history-update-check.sh]: Yes, since it's its own test case, I figured it could have some more output. :)

**Pull request merged**